qiskit/primitives/base_estimator.py

```diff
@@ -0,0 +1,233 @@
+and users can specify pairs of quantum circuits and observables
+to estimate the expectation values.
+
+An estimator is initialized with the following elements.
+
+* quantum circuits (:math:`\psi_i(\theta)`): list of (parameterized) quantum circuits
+  (a list of :class:`~qiskit.circuit.QuantumCircuit`))
+
+* observables (:math:`H_j`): a list of :class:`~qiskit.quantum_info.SparsePauliOp`.
+
+The estimator is called with the following inputs.
+
+* circuit indexes: a list of indexes of the quantum circuits.
+
+* observable indexes: a list of indexes of the observables.
+
+* parameters: a list of parameters of the quantum circuits.
+  (:class:`~qiskit.circuit.parametertable.ParameterView` or
+  a list of :class:`~qiskit.circuit.Parameter`).
+
+* parameter values (:math:`\theta_k`): list of sets of values
+  to be bound to the parameters of the quantum circuits.
+  (list of list of float)
+
+The output is an EstimatorResult which contains a list of expectation values plus
+optional metadata like confidence intervals for the estimation.
+
+.. math::
+
+    \langle\psi_i(\theta_k)|H_j|\psi_i(\theta_k)\rangle
+
+
+The estimator object is expected to be `close()` d after use or
+accessed inside "with" context
+and the objects are called with parameter values and run options
+(e.g., ``shots`` or number of shots).
+
+Here is an example of how estimator is used.
+
+.. code-block:: python
+
+    from qiskit.circuit.library import RealAmplitudes
+    from qiskit.quantum_info import SparsePauliOp
+
+    psi1 = RealAmplitudes(num_qubits=2, reps=2)
+    psi2 = RealAmplitudes(num_qubits=2, reps=3)
+
+    params1 = psi1.parameters
+    params2 = psi2.parameters
+
+    H1 = SparsePauliOp.from_list([("II", 1), ("IZ", 2), ("XI", 3)])
+    H2 = SparsePauliOp.from_list([("IZ", 1)])
+    H3 = SparsePauliOp.from_list([("ZI", 1), ("ZZ", 1)])
+
+    with Estimator([psi1, psi2], [H1, H2, H3], [params1, params2]) as e:
+        theta1 = [0, 1, 1, 2, 3, 5]
+        theta2 = [0, 1, 1, 2, 3, 5, 8, 13]
+        theta3 = [1, 2, 3, 4, 5, 6]
+
+        # calculate [ <psi1(theta1)|H1|psi1(theta1)> ]
+        psi1_H1_result = e([0], [0], [theta1])
+        print(psi1_H1_result)
+
+        # calculate [ <psi1(theta1)|H2|psi1(theta1)>, <psi1(theta1)|H3|psi1(theta1)> ]
+        psi1_H23_result = e([0, 0], [1, 2], [theta1]*2)
+        print(psi1_H23_result)
+
+        # calculate [ <psi2(theta2)|H2|psi2(theta2)> ]
+        psi2_H2_result = e([1], [1], [theta2])
+        print(psi2_H2_result)
+
+        # calculate [ <psi1(theta1)|H1|psi1(theta1)>, <psi1(theta3)|H1|psi1(theta3)> ]
+        psi1_H1_result2 = e([0, 0], [0, 0], [theta1, theta3])
+        print(psi1_H1_result2)
+
+        # calculate [ <psi1(theta1)|H1|psi1(theta1)>,
+        #             <psi2(theta2)|H2|psi2(theta2)>,
+        #             <psi1(theta3)|H3|psi1(theta3)> ]
+        psi12_H23_result = e([0, 0, 0], [0, 1, 2], [theta1, theta2, theta3])
```

Comment:
Because H1, H2 and H3 are involved in.
```suggestion
        psi12_H123_result = e([0, 0, 0], [0, 1, 2], [theta1, theta2, theta3])
```